### ISSUE TITLE
IP Ban bypass

### ISSUE THREAD

**Issue by Minecraft-Classic-Server-Archive** [ClassiCube/MCGalaxy#673]
Joining with an admin's name blocks the ability to IP Ban the user as the server thinks that that is their name even if they dont log in
You can also join with multiple accounts of the same name by changing the case (like Hedwig7s and hedwig7s can join at the same time)

**Comment by UnknownShadow200** [comment]
Are you running with name verification disabled? I am unable to reproduce when running when name verification enabled
![image](https://user-images.githubusercontent.com/6509348/146910358-81df4c2b-1fa7-4fdd-b494-9cb073a26389.png)

**Comment by Goodlyay** [comment]
Furthermore, you should follow this guide so that verify names can be turned on, as it is an essential part of running a server: https://f.classicube.net/topic/1590-how-to-set-up-authentication-for-players-on-betacraft/?tab=comments#comment-8582

**Comment by Minecraft-Classic-Server-Archive** [comment]
> Are you running with name verification disabled? I am unable to reproduce when running when name verification enabled ![image](https://user-images.githubusercontent.com/6509348/146910358-81df4c2b-1fa7-4fdd-b494-9cb073a26389.png)

No but it is still an issue
Dunno why but xDeathCon wont turn it on

**Comment by Minecraft-Classic-Server-Archive** [comment]
Also I mean same rank
OFC a higher rank can ban em

**Comment by Minecraft-Classic-Server-Archive** [comment]
But it makes moderation as an operator annoying as hell 
I think impersonating somebody with SuperOP and Nobody also does it

**Comment by Minecraft-Classic-Server-Archive** [comment]
my server does have name verification but if there is the option for name verification to be disabled it should be unexploitable (thats what passwords and blocking a 2nd person with the same name are for)

**Comment by Minecraft-Classic-Server-Archive** [comment]
to trigger the glitch 2 people with different ips must connect on the same admin account (same caps or not)
Then if someone tries to ban em if they have the same rank or lower it blocks it because "Admin has the same IP" even if its the impostors ip

**Comment by UnknownShadow200** [comment]
In the above example, operator UnknownShadow is trying to ban the IP that superop UnknownShadow200 failed to login from, so it's behaving as expected (with name verification enabled)

---

I have made clone detection case insensitive with c5913ad8dd0e5da4de3db953e624518c5e78f625

---

To properly implement this when name verification is disabled would require some changes (just delaying 'last ip' update until after /pass verification is insufficient, would require adding and tracking a separate 'last authenticated IP' for players)

And I'm not really a fan of spending time implementing that, when it's already made clear to a server owner that running with name verification off is a bad idea as it is insecure and dangerous
![image](https://user-images.githubusercontent.com/6509348/146926442-da53fe61-eb07-40c4-84f6-ee1937245859.png)

**Comment by Minecraft-Classic-Server-Archive** [comment]
![161746610 jpg](https://user-images.githubusercontent.com/96392815/146934445-7f780ea5-b3cc-42d4-8e5f-6ce89fa651bc.png)
> In the above example, operator UnknownShadow is trying to ban the IP that superop UnknownShadow200 failed to login from, so it's behaving as expected (with name verification enabled)
> 
> I have made clone detection case insensitive with [c5913ad](https://github.com/UnknownShadow200/MCGalaxy/commit/c5913ad8dd0e5da4de3db953e624518c5e78f625)
> 
> To properly implement this when name verification is disabled would require some changes (just delaying 'last ip' update until after /pass verification is insufficient, would require adding and tracking a separate 'last authenticated IP' for players)
> 
> And I'm not really a fan of spending time implementing that, when it's already made clear to a server owner that running with name verification off is a bad idea as it is insecure and dangerous ![image](https://user-images.githubusercontent.com/6509348/146926442-da53fe61-eb07-40c4-84f6-ee1937245859.png)

How would just delaying the last ip change until after /pass be insufficient? If it is because it would make non-admin accounts not update couldn't you do something like this?
 (gonna write it in roblox lua)
![image](https://user-images.githubusercontent.com/96392815/146934076-b3cbd52a-4263-4b0e-a378-6ff48deb8b1f.png)

**Comment by Minecraft-Classic-Server-Archive** [comment]
Added it as an image cause github was messing up my formatting

**Comment by forkiesassds** [comment]
> Dunno why but xDeathCon wont turn it on

Probably it's so they can allow piracy on their server.

**Comment by Minecraft-Classic-Server-Archive** [comment]
bruh

**Comment by UnknownShadow200** [comment]
> How would just delaying the last ip change until after /pass be insufficient?

If a suspicious login attempt was reported for a `/pass` protected account, staff wouldn't be able to use `/info` to see the IP that the suspicious login attempt occurred from anymore (last IP)

**Comment by Minecraft-Classic-Server-Archive** [comment]
Couldn't you have a secondary ip stored and when you do /info it gives:
"Last IP: 111.222.333"
"Last Verified IP: 222.333.444" and instead of blocking an ip ban because
the ip matches the last known ip of an admin block it because it matches
the last verified ip?

**Comment by UnknownShadow200** [comment]
right, that's what I meant before about would also need a separate 'last authenticated IP'
